lib/js/app/modules/queries/types.ts

```diff
@@ -143,4 +152,5 @@ export type QueriesActions =
   | RunQuerySuccessAction
   | GetSavedQueriesAction
   | GetSavedQueriesSuccessAction
-  | GetSavedQueriesErrorAction;
+  | GetSavedQueriesErrorAction
+  | SetQueryLimitReachedAction;
```

Comment:
Is that action invoked anywhere ? I believe to cover use case with execution queries limit reached. You should dispatch that action in error handler for run query saga.

Reply:
```
function* runQuery(action: RunQueryAction) {
  try {
    const {
      payload: { body },
    } = action;
    const client = yield getContext('keenClient');
    const responseBody = yield client.query(body);

    yield put(runQuerySuccess(responseBody));
  } catch (error) {
    console.log('error', error);
    yield put(runQueryError(error));
  }
}

```